src/Fixer/Import/GlobalNamespaceImportFixer.php

```diff
@@ -706,7 +706,7 @@ private function traverseDocBlockTypes(DocBlock $doc, callable $callback): bool
             foreach ($types as $i => $fullType) {
                 $newFullType = $fullType;
 
-                Preg::matchAll('/[\\\\\w]+/', $fullType, $matches, PREG_OFFSET_CAPTURE);
+                Preg::matchAll('/[\\\\\w]+(?![\\\\\w:])/', $fullType, $matches, PREG_OFFSET_CAPTURE);
```

Comment:
I've created a playground: [v1](https://regex101.com/r/Q8K4U2/1) shows current version, [v2](https://regex101.com/r/Q8K4U2/2) represents your fix, [v3](https://regex101.com/r/Q8K4U2/3) is my suggestion:

```suggestion
                Preg::matchAll('/[\\\\\w\-]+(?![\\\\\w:])/', $fullType, $matches, PREG_OFFSET_CAPTURE);
```

That would properly match non-native types like `class-string` - these can't be imported, but I think they should be matched anyway, so any logic later can decide what to do with it.

Reply:
> these can't be imported, but I think they should be matched anyway, so any logic later can decide what to do with it.

I don't get what "any logic late" could do anything as these cannot be imported.

Reply:
Yeah, you're right, if it was more generic regex, used in other fixers, then it would matter. Here it's enough to exclude array shape's keys 👍. The point with actual parser stands 😉.